### PR TITLE
[core] Adding the option to rename the SES output record

### PR DESCRIPTION
**Pull request by Meliairon**
to: @airbnb/streamalert-maintainers
size: small

## Background

Requirement from our security team that the json records contain additional information in the name of the file

## Changes

* [core] Added the ability to specify the name of the record.json file in the SES output
* [testing] Modified the unit tests to test this new functionality and to take into account the changes

## Testing

* ran pylint ./tests/scripts/pylint.sh
* ran unit tests ./tests/scripts/unit_tests.sh

### CODE REVIEW

File: streamalert/alert_processor/outputs/aws.py [hunk 1]
```diff
@@ -550,7 +550,7 @@ def _build_email(cls, alert, publication, creds):
         default_subject = "{} triggered alert {}".format(
             alert.rule_name, alert.alert_id
         )
-        default_body = "Please review the attached record.json"
+        default_body = "Please review the attached json record"
```
Comment (jack1902) [inline streamalert/alert_processor/outputs/aws.py]: You can over-ride this on a publisher, same applies to the record. Just drop it with `publication['@aws-ses.attach_record'] = False` and then specify it as your own attachment with your own Body

File: streamalert/alert_processor/outputs/aws.py [hunk 2]
```diff
@@ -566,7 +566,8 @@ def _build_email(cls, alert, publication, creds):
         # Attach the record to the email
         if publication.get("@aws-ses.attach_record", True):
             record = json.dumps(alert.record, sort_keys=True, indent=2)
-            msg = cls._add_attachment(msg, "record.json", record)
+            record_name = publication.get("@aws-ses.record_name", "record")
+            msg = cls._add_attachment(msg, f"{record_name}.json", record)
```
Comment (jack1902) [inline streamalert/alert_processor/outputs/aws.py]: f-strings are currently not used across StreamAlert. So not sure if the introduction will be accepted or not

File: streamalert/alert_processor/outputs/aws.py [hunk 3]
```diff
@@ -621,6 +622,8 @@ def _dispatch(self, alert, descriptor):
             - @aws-ses.attach_record (bool):
                 True (default): Attach the alert.record to the email
                 False: Don't attach the alert.record to the email
+            - @aws-ses.record_name (str):
```
Comment (jack1902) [inline streamalert/alert_processor/outputs/aws.py]: As per the other comment,

You can pass in a dict like so:
```python
# Assuming presence of Alert
# publication = {}
# publication['@aws-ses.attachements'] = {"FILE_NAME": "FILE_CONTENTS"}

# Remove the default record.json attachment
publication['@aws-ses.attach_record'] = False

# Add the attachment with a name
publication['@aws-ses.attachments'] = {"record.json": json.dumps(alert.record)}
```

File: streamalert/alert_processor/outputs/aws.py (hunk 2 above)
Comment (Meliairon) [reply streamalert/alert_processor/outputs/aws.py]: If this is not the accepted convention then I can easily change this

File: streamalert/alert_processor/outputs/aws.py (hunk 1 above)
Comment (Meliairon) [reply streamalert/alert_processor/outputs/aws.py]: This does seem to be a rather roundabout way of just renaming the default record, removing it and readding it with a different name.

File: streamalert/alert_processor/outputs/aws.py [hunk 4]
```diff
@@ -566,7 +566,8 @@ def _build_email(cls, alert, publication, creds):
         # Attach the record to the email
         if publication.get("@aws-ses.attach_record", True):
             record = json.dumps(alert.record, sort_keys=True, indent=2)
-            msg = cls._add_attachment(msg, "record.json", record)
+            record_name = publication.get("@aws-ses.record_name", "record")
```
Comment (jack1902) [inline streamalert/alert_processor/outputs/aws.py]: in line with how other "defaults" are configured. I would move `"record"` to `default_record_name = "record"` and then reference this on this line (similar to the default_body etc)

File: tests/unit/streamalert/alert_processor/outputs/test_aws.py [hunk 5]
```diff
@@ -356,6 +356,26 @@ def test_add_single_attachment(self):
             # Raise an error if no payload of type MIMEApplication is found
             raise AssertionError
 
+    def test_different_attachment_name(self):
```
Comment (jack1902) [inline tests/unit/streamalert/alert_processor/outputs/test_aws.py]: i'd rename to `test_record_name_override` or something relevant to the record, as attachments are already tested by the previous piece

**Comment by Ryxias** [comment]
Closing outdated PR;

For this PR I'd just include this in your company-specific private repo, using Alert Publishers. You can then OSS the publishers in the StreamAlert sample code repo... which we're working on!

**Pull request closed without merging**